src/main/java/net/dries007/tfc/common/blocks/plant/KelpTreeBlock.java

```diff
@@ -0,0 +1,167 @@
+        .setValue(getFluidProperty(), getFluidProperty().keyFor(Fluids.EMPTY)));
+    }
+
+    @Override
+    public BlockState getStateForPlacement(BlockItemUseContext context)
+    {
+        return this.getStateForPlacement(context.getLevel(), context.getClickedPos());
+    }
+
+    public BlockState getStateForPlacement(IBlockReader world, BlockPos pos)
+    {
+        Block downBlock = world.getBlockState(pos.below()).getBlock();
+        Block upBlock = world.getBlockState(pos.above()).getBlock();
+        Block northBlock = world.getBlockState(pos.north()).getBlock();
+        Block eastBlock = world.getBlockState(pos.east()).getBlock();
+        Block southBlock = world.getBlockState(pos.south()).getBlock();
+        Block westBlock = world.getBlockState(pos.west()).getBlock();
+        return this.defaultBlockState()
+            .setValue(DOWN, downBlock.is(TFCTags.Blocks.KELP_TREE) || downBlock.is(TFCTags.Blocks.SEA_BUSH_PLANTABLE_ON))
+            .setValue(UP, upBlock.is(TFCTags.Blocks.KELP_TREE))
+            .setValue(NORTH, northBlock.is(TFCTags.Blocks.KELP_TREE))
+            .setValue(EAST, eastBlock.is(TFCTags.Blocks.KELP_TREE))
+            .setValue(SOUTH, southBlock.is(TFCTags.Blocks.KELP_TREE))
+            .setValue(WEST, westBlock.is(TFCTags.Blocks.KELP_TREE));
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public BlockState updateShape(BlockState stateIn, Direction facing, BlockState facingState, IWorld worldIn, BlockPos currentPos, BlockPos facingPos)
+    {
+        if (!stateIn.canSurvive(worldIn, currentPos))
+        {
+            worldIn.getBlockTicks().scheduleTick(currentPos, this, 1);
+            updateFluid(worldIn, stateIn, currentPos);
+            return stateIn;
+        }
+        else
+        {
+            updateFluid(worldIn, stateIn, currentPos);
+            boolean flag = facingState.is(TFCTags.Blocks.KELP_TREE) || (facing == Direction.DOWN && facingState.is(TFCTags.Blocks.SEA_BUSH_PLANTABLE_ON));
+            return stateIn.setValue(PROPERTY_BY_DIRECTION.get(facing), flag);
+        }
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public void tick(BlockState state, ServerWorld worldIn, BlockPos pos, Random rand)
+    {
+        if (!state.canSurvive(worldIn, pos))
+        {
+            worldIn.destroyBlock(pos, true);
+        }
+    }
+
+    @Override
+    public void playerWillDestroy(World worldIn, BlockPos pos, BlockState state, PlayerEntity player)
+    {
+        updateFluid(worldIn, state, pos);
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public boolean canSurvive(BlockState state, IWorldReader worldIn, BlockPos pos)
+    {
+        BlockState belowState = worldIn.getBlockState(pos.below());
+        //boolean attachedVertically = !KelpTreeFlowerBlock.isEmptyWaterBlock(worldIn, pos.above()) && !KelpTreeFlowerBlock.isEmptyWaterBlock(worldIn, pos.below());
```

Comment:
Leaving block commented code is bad. Either there's a good explanation and reason for keeping this here (in which case also include said comment), or remove the commented code.